jax/experimental/shard_map.py

```diff
@@ -1634,7 +1634,7 @@ def _all_mesh_names(mesh: Mesh) -> set[AxisName]:
   names = {n for frame in stack
            if (ns := frame.payload.get('spmd_axis_name', ())) is not None
            for n in ns}
-  return set(mesh.axis_names) - names
+  return tuple(sorted(set(mesh.axis_names) - names))
```

Comment:
Instead of sorting, could we inherit the axis order from `mesh.axis_names`, like
```suggestion
  return tuple(name for name in mesh.axis_names if name not in names)
```

Reply:
Done.